### ISSUE TITLE
Variable initialization problem

### ISSUE THREAD

**Issue by 3rd3** [tensorflow/tensorflow#4408]
Hi, I am getting a error "Attempting to use uninitialized value" on tf.initialize_all_variables when I try to incorporate the shape of a tensor via tf.shape in a loss function that depends on a dynamically shaped input tensor, and in between, on the variable that does not get initialized and on tf.scan. I am trying to get a minimal working example of the bug, but I couldn't reproduce it yet, so I thought I would ask whether this or a similar issue is already known before I continue. The only reproducible step is that the initialization works when I replace tf.to_float(tf.shape(x))[0] in my loss function with tf.constant(1.0), which makes me believe it has something to do with tf.shape or perhaps with the tf.scan.

**Comment by andydavis1** [comment]
Without a reproducible test case, it would probably be best to first ask around on StackOverflow and see if others can answer your question. If you cant get a good answer and StackOverflw, and if you can get a working example that reproduces the problem, please submit it here.
